### PR TITLE
test: improve errors package coverage from 55% to 95%

### PR DESCRIPTION
**Pull request by TheTimbob**
## Summary
- Add tests for `LogError` covering all severity levels (info, warning, error, critical), nil input, and handler context
- Add tests for `Is` and `As` delegation to standard library
- Add tests for `WithErr` copy independence
- Add missing constructor coverage for `Unauthorized` and `LoginFailed`
- Errors package coverage: 55.4% → 94.6%

## Test plan
- [x] `go test ./internal/errors/ -v` — all pass
- [x] `golangci-lint run ./internal/errors/` — 0 issues
- [x] No behavioral changes — test-only PR

### CODE REVIEW

File: internal/errors/errors_test.go [hunk 1]
```diff
@@ -132,3 +135,101 @@ func TestSeverityConstructors(t *testing.T) {
 		})
 	}
 }
+
+func TestLogError(t *testing.T) {
```
Comment (TheTimbob) [inline internal/errors/errors_test.go:154]: **[warning]** All five `TestLogError` subtests call `apperrors.LogError` but make no assertions — they only verify the function doesn't panic. The most significant gap is the "logs critical error with stack trace" subtest: if the `debug.Stack()` call in `handler.go:43` were accidentally removed, this test would still pass.

Consider capturing log output and asserting on it:

```go
func TestLogError(t *testing.T) {
    t.Parallel()

    t.Run("logs critical error with stack trace", func(t *testing.T) {
        t.Parallel()

        var buf bytes.Buffer
        log.SetOutput(&buf)
        t.Cleanup(func() { log.SetOutput(os.Stderr) })

        err := apperrors.PanicRecovered(errors.New("panic"))
        apperrors.LogError(err)

        got := buf.String()
        if !strings.Contains(got, "[STACK]") {
            t.Errorf("expected stack trace in log output, got: %s", got)
        }
        if !strings.Contains(got, "PANIC_RECOVERED") {
            t.Errorf("expected error code in log output, got: %s", got)
        }
    })
}
```

Note: parallel subtests sharing `log.SetOutput` need a mutex around the buffer write; you may want to make the subtests sequential (drop the inner `t.Parallel()`) or use a `sync.Mutex`-wrapped writer.
